mobile/lib/features/trade/domain/trade_values.dart

```diff
@@ -113,7 +132,13 @@ class TradeValues {
   _recalculateQuantity() {
     Usd? quantity =
         tradeValuesService.calculateQuantity(price: price, margin: margin, leverage: leverage);
-    this.quantity = quantity;
+    this.quantity = quantity ?? Usd.zero();
+  }
+
+  // Note, this only works because it is only called from [`updatePriceAndQuantity`], which is only
+  // called if the max quantity lock is set.
+  _recalculateContracts() {
+    contracts = quantity + openQuantity;
```

Comment:
Wouldn't it be safer to do this from inside `_recalculateQuantity()`, or whenever `quantity` is being updated?

Reply:
I think it wouldn't be safer, as we want to do this only if the max quantity lock is set. otherwise that would be wrong. As mentioned in the comment, this only works because its only called from the `updatePriceAndQuantity` which is only called if the `maxQuantityLock` is set.

Reply:
Actually after some testing, it turns out it wasn't working as expected. We achieve the original intention by simply setting the contracts to the maxQuantity after every price update.